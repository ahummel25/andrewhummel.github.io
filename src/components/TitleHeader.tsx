import React, { FC } from 'react';
import styled from '@emotion/styled';

import { breakpoints as bp } from '../styles/variables';

const Header = styled.h3`
  margin: 0 auto;
  background-color: rgb(0, 128, 115);
  font-weight: 100;
  padding: 30px 40px;
  margin-bottom: 60px;
  text-transform: uppercase;

  @media (max-width: ${bp.lg}px) {
    padding: 20px 30px;
    margin-bottom: 40px;
  }
`;

const TitleHeader: FC<{ title: string }> = ({ title }): JSX.Element => (
  <>
    <Header className="page-title white-text">{title}</Header>
  </>
);

export default TitleHeader;
