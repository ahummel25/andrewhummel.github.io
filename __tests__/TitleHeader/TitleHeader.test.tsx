import React from 'react';
import { create } from 'react-test-renderer';

import TitleHeader from '../../src/components/TitleHeader';

describe('TitleHeader', () => {
  it('renders correctly', () => {
    const tree = create(<TitleHeader title="Mock Title Header" />);

    expect(tree).toBeDefined();
  });
});
