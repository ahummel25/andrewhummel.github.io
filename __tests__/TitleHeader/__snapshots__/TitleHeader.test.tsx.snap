// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleHeader renders correctly 1`] = `
.emotion-0 {
  font-family: Roboto,sans-serif;
  margin: 0 auto;
  background-color: rgb(0, 128, 115);
  font-weight: 100;
  padding: 30px 40px;
  margin-bottom: 60px;
  text-transform: uppercase;
}

@media (max-width: 992px) {
  .emotion-0 {
    padding: 20px 30px;
    margin-bottom: 40px;
  }
}

<h3
  className="page-title white-text emotion-0"
>
  Mock Title Header
</h3>
`;
